freqtrade/persistence/models.py

```diff
@@ -57,6 +58,8 @@ def init_db(db_url: str) -> None:
     Trade.query = Trade._session.query_property()
     Order.query = Trade._session.query_property()
     PairLock.query = Trade._session.query_property()
+    KeyValue._session = scoped_session(sessionmaker(bind=engine, autoflush=True))
+    KeyValue.query = KeyValue._session.query_property()
```

Comment:
i'm ... skeptical about this (this could end up deadlocking the bot) in odd cases

The assumption with a 2nd session is that no KeyValue object will ever be touched by the "main" session - as all access goes through `trade.set_kval()` and `trade.get_kval()`.
The way the models are currently setup, that's an incorrect assumption as sqlalchemy lazy-loads relationships by default.

Put that into your `populate_*` methods (or any other method, really)

```python
# Use any filter you like - but you must use .get_trades for this to get a non-detached trade object
trade = Trade.get_trades([Trade.id == 1])

for kvals in trade.keyvalues:   # This will lazy-load this array via an additional select
    print(kvals)
```

Now i'm fully aware that through this method, you could also all trade data (this is a full, rw trade object) - but if users do this, one must assume it's intentional and if it breaks, it's not freqtrade's fault.